stepfunctions-local-patch/cloud/localstack/RegionAspect.java

```diff
@@ -97,4 +104,90 @@ public StateExecutor aroundCreateExecutor(ProceedingJoinPoint joinPoint) throws
+
+        if (parts.length == 1) {
+            // http://localhost:4566/....
+            return parts[0].startsWith("localhost");
+        } else if (parts.length == 6 && Objects.equals(parts[4], "localstack")) {
+            // http(s)://<restapi-id>.execute-api.us-east-1.localhost.localstack.cloud/...
+            return true;
+        } else if (parts.length == 5 && Objects.equals(parts[3], "localstack")) {
+            // http(s)://<restapi-id>.execute-api.localhost.localstack.cloud/...
+            return true;
+        }
+
+        return (boolean) joinPoint.proceed(joinPoint.getArgs());
+    }
+
+    /**
+     * This just needs to return *some* string for now (won't be used later on)
+     */
+    @Around("execution(* com.amazonaws..ApiGatewayInvokeCaller.parseRegionFromEndpoint(..))")
+    public String aroundParseRegionFromEndpoint(ProceedingJoinPoint joinPoint) throws Throwable {
+        return "replaceme";
+    }
+
+    /**
+     *  We're using the config in the arguments to set the region argument
+     *  This would otherwise be the "replaceme" string from above
+     */
+    @Around("execution(* com.amazonaws..ApiGatewayInvokeCaller.signRequestWithSigV4IfNecessary(..))")
+    public String aroundSignRequestWithSigV4IfNecessary(ProceedingJoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        Config config = (Config) joinPoint.getArgs()[3];
+        args[2] = config.getRegion();
+        return (String) joinPoint.proceed(args);
+    }
+
+    /**
+     * Patches the generated URLs for the localstack specific URLS (localhost  or ... localhost.localstack.cloud)
+     */
+    @Around("execution(* com.amazonaws..ApiGatewayInvokeCaller.createUrl(..))")
+    public URI aroundCreateUrl(ProceedingJoinPoint joinPoint) throws Throwable {
+        ApiGatewayInvokeRequest invokeRequest = (ApiGatewayInvokeRequest) joinPoint.getArgs()[0];
+        String endpoint = invokeRequest.getApiEndpoint();
+
+        if (endpoint.contains("localhost:4566")) {
+            // remove region for localstack
+            StringBuilder urlBuilder = (new StringBuilder("http://")).append(endpoint);
+            if (invokeRequest.getStage() != null) {
+                urlBuilder.append("/").append(SdkHttpUtils.urlEncodeIgnoreSlashes(invokeRequest.getStage()));
+            }
+
+            if (invokeRequest.getPath() != null) {
+                String path = StringUtils.removeStart(invokeRequest.getPath(), "/");
+                path = StringUtils.removeEnd(path, "/");
+                urlBuilder.append("/").append(SdkHttpUtils.urlEncodeIgnoreSlashes(path));
+            }
+
+            return URI.create(urlBuilder.toString());
+        } else {
+            // remove region for localstack
+            int start = endpoint.indexOf(".", endpoint.indexOf(".") + 1);
+            int end = endpoint.indexOf(".", start + 1);
+            String endpointWithoutRegion = endpoint.substring(0, start) + endpoint.substring(end);
```

Comment:
nit: perhaps a regex would be a bit easier for this replacement?

Reply:
If you insist, we can change this otherwise I'd leave this since it's fairly stable